### ISSUE TITLE
Proposed: A --sandbox option that allows a resource folder for images

### ISSUE THREAD

**Issue by kchristman54** [jgm/pandoc#7679]
Point 3 of pandoc's [A note on security](https://pandoc.org/MANUAL.html#a-note-on-security), says: 
> An untrusted attacker could exploit this to view the contents of non-image files on the file system. (Using the `--sandbox` option can protect against this threat, but will also prevent including images in these formats.)

For those who run pandoc on user-generated (therefore **untrusted!**) content on a web-server, the `--sandbox` option is of interest. But currently it prevents including images.

What if we could opt-in to a less strict sandbox approach, where the working/resource/data-dir directory (and any sub-directories) was **exclusively** allowed? This would open up sandboxing to those who need images in their documents, and still prevent inappropriate access to other parts of the file-system, like the OS or web-server files. In this proposed mode, pandoc would ensure that any requested resource (image) would legitimately be in the working/resource/data-dir folder, preventing a Directory Traversal Attack ([1](https://en.wikipedia.org/wiki/Directory_traversal_attack), [2](https://owasp.org/www-community/attacks/Path_Traversal#), [3](https://portswigger.net/web-security/file-path-traversal)).   

Relevant: #5045

**Comment by jgm** [comment]
The way `--sandbox` is implemented, readers *cannot* do IO (anywhere).  They don't have a type that allows this.  (That is good, because we can really be *sure* that they don't.)

So, the best we can do is to load files that the reader will need into an ersatz file system before calling the reader.  But then we need to know *what* to load before the reader is called.  I think preloading everything in the working directory (or even all the images found there or below) is going to be a bad idea.

**Comment by theZappr** [comment]
Hi All,

Awesome project BTW :star:

We have been using pandoc 3.0.1 `--sandbox` for it's security benefits fine with `docx -> html` but since trialing with `-t chunkedhtml` it fails on the exports because it can not find the (exported!) images in the "resource path".

No matter what I put into `--resource-path` (which I never needed before) it doesn't resolve it. This was not a problem before trying `chunkedhtml`.   The only way I can find to get `chunkedhtml` to export successfully is to _not_ use `--sandbox` . However, that could compromise the system so is not a viable option, in our case. Who would trust a docx under any circumstances :P

It sounds to me that this is related to this ticket. Or should I raise another? I could, of course, be doing something hopelessly wrong but I don't think so.

**Comment by jgm** [comment]
Yes, that's the sort of thing this issue is meant to help with.  Pandoc's chunkedhtml writer would need to use IO to read the images, and `--sandbox` completely blocks this.

**Comment by philbo** [comment]
is it possible to achieve _some_ level of security by using filters when sandbox mode is disabled? i.e. filter specific things like `include`? If so, does anyone know of any existing filters that do this?

**Comment by jgm** [comment]
Sorry, `.. include` (in RST for example) would be processed by the parser, before a filter is run.
